docs/pages/interfaces.rst

```diff
@@ -3,8 +3,234 @@
+
+Mappable
+--------
+
+Something is considered mappable if we can ``map`` it using a function,
+generally ``map`` is a method that accepts a function.
+
+An example in this library is :class:`Maybe <returns.maybe.Maybe>`,
+that implements the ``Mappable`` interface:
+
+.. code:: python
+
+  >>> from returns.maybe import Maybe, Some
+
+  >>> def mappable(string: str) -> str:
+  ...     return string + '!'
+
+  >>> maybe_str: Maybe[str] = Some('example')
+  >>> assert maybe_str.map(mappable) == Some('example!')
+
+:class:`Mappable <returns.interfaces.mappable.MappableN>` interface help us to
+create our own mappable container like :class:`Maybe <returns.maybe.Maybe>`.
+
+.. code:: python
+
+  >>> from typing import Any, Callable, TypeVar
+
+  >>> from returns.interfaces.mappable import Mappable1
+  >>> from returns.primitives.hkt import Kind1
+
+  >>> _NumberType = TypeVar('_NumberType')
+  >>> _NewNumberType = TypeVar('_NewNumberType')
+
+  >>> class Number(
+  ...     Kind1['Number', _NumberType],
+  ...     Mappable1[_NumberType],
+  ... ):
+  ...     def __init__(self, inner_value: _NumberType):
+  ...         self._inner_value = inner_value
+  ...
+  ...     def map(  # This method is required by Mappable
+  ...         self,
+  ...         function: Callable[[_NumberType], _NewNumberType]
+  ...     ) -> 'Number[_NewNumberType]':
+  ...         return Number(function(self._inner_value))
+  ...
+  ...     def __eq__(self, other: Any) -> bool:  # Required to check the identity law
+  ...         if not isinstance(self, type(other)):
+  ...             return False
+  ...         return self._inner_value == other._inner_value
+
+With our ``Number`` mappable class we can compose easily math functions with it.
+
+.. code:: python
+
+  >>> def my_math_function(number: int) -> int:
+  ...     return number - 1
+
+  >>> number: Number[int] = Number(-41)
+  >>> assert number.map(my_math_function).map(abs) == Number(42)
+
+Laws
+~~~~
+
+To make sure your mappable implementation is right, you can apply the
+Mappable laws on it to test.
+
+1. **Identity Law:** When we pass the identity function to the map method,
+the mappable has to be the same, unaltered.
+
+.. code:: python
+
+  >>> from returns.functions import identity
+
+  >>> mappable_number: Number[int] = Number(1)
+  >>> assert mappable_number.map(identity) == Number(1)
+
+2. **Associative Law**: Given two functions, `x` and `y`, calling the map
+method with `x` function and after that calling with `y` function must have the
+same result if we compose them together.
+
+.. code:: python
+
+  >>> def add_one(number: int) -> int:
+  ...     return number + 1
+
+  >>> def multiply_by_ten(number: int) -> int:
+  ...     return number * 10
+
+  >>> mappable_number: Number[int] = Number(9)
+  >>> assert mappable_number.map(add_one).map(multiply_by_ten) == mappable_number.map(lambda value: multiply_by_ten(add_one(value)))
```

Comment:
What do you think, should we use `compose` here?

Reply:
I think so, maybe it'll be more clear to understand the `law`!!